.github/actions/merge-develop-and-set-up-dependencies/action.yml

```diff
@@ -31,6 +35,25 @@ runs:
         key: ${{ runner.os }}-proddeps-${{ hashFiles('/home/runner/work/.pyversion', 'requirements_dev.txt', 'scripts/install_python_dev_dependencies.py') }}
         path: |
           /opt/hostedtoolcache/Python/3.9.20/x64
+
+    # TODO (#22567): Fix pip not working properly when fetched from cache.
+    - name: Check cached pip installation and fix if needed
```

Comment:
Per your previous comment, it sounds like this might not be the cached pip installation after all (in some cases), so we might need to remove "cached" from this step.

Reply:
Done.